susemanager-utils/susemanager-sls/src/_states/uyuni_users.py

```diff
@@ -0,0 +1,459 @@
+
+    def manage(self, name: str, password: str, email: str, first_name: str = "", last_name: str = "",
+               org: str = "", roles: Optional[List[str]] = None) -> Dict[str, Any]:
+        """
+        Manage user with the data. If anything of the initial data is updated, is going to be added.
+
+        :param name: UID of the user
+        :param password: Password for the user
+        :param email: Email of the user
+        :param first_name: First name (optional)
+        :param last_name: Second name (options)
+        :param org: Organisation
+        :param roles: list of roles
+
+        :return:
+        """
+        result: Optional[bool] = None
+        ret = {
+            'name': name,
+            'changes': {},
+            'result': result,
+            'comment': "",
+        }
+
+        existing_user = None
+        try:
+            for user in self.users.get_all_users():
+                if user.get("login") == name:
+                    existing_user = self.users.get_user(name)
+                    break
+        except UyuniUsersException as exc:
+            ret["comment"] = "Error manage user '{}': {}".format(name, exc)
+            ret["result"] = False
+            log.error(ret["comment"])
+        else:
+            if existing_user is None:
+                changes = {"uid": name, "password": password, "email": email,
+                           "first_name": first_name, "last_name": last_name}
+                self.create(**changes)
+                changes["password"] = "(hidden)"
+
+                ret["changes"] = changes
+                ret["comment"] = "Added new user {}".format(name)
+                ret["result"] = True
+            else:
+                # Changes
+                ret["comment"] = "No changes has been done"
+
+        return ret
+
+
+def __virtual__():
+    """
+    Provide Uyuni Users state module.
+
+    :return:
+    """
+    return __virtualname__ if RPCClient.init(pillar=__pillar__) is not None else False
+
+
+# Salt exported API
+
+
+def user_present(name, password, email, first_name, last_name, org="", roles=None):
+    """
+    Ensure Uyuni user present.
+
+    :param name: Uyuni user name
+    :param password: Password (should be empty in case of PAM auth)
+    :param email: Email for the user
+    :param first_name: First name
+    :param last_name: Second (last) name
+    :param org: Organisation name
+    :param roles: List of Uyuni roles that user has to have.
+
+    :raises: UyuniException if roles aren't correctly spelt.
+    :return: dictionary for Salt communication protocol
+    """
+    return UyuniUsers().manage(name=name, password=password, email=email, first_name=first_name,
+                               last_name=last_name, org=org, roles=roles)
+
+
+def user_absent(name):
+    """
+    Remove Uyuni user.
+
+    :param name: Uyuni user name
+
+    :return: dictionary for Salt communication protocol
+    """
+    return UyuniUsers().delete(name)
```

Comment:
Is this going to be idempotent? At a first glance, it will throw an error if the user is already absent?